apiserver/test/e2e/cluster_server_e2e_test.go

```diff
@@ -0,0 +1,256 @@
+package e2e
+
+import (
+	"net/http"
+	"testing"
+	"time"
+
+	rayv1api "github.com/ray-project/kuberay/ray-operator/apis/ray/v1"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+// TestClusterServerEndpoints sequentially iterates over the endpoints of the Clusters endpoints
+func TestClusterServerEndpoints(t *testing.T) {
+	tCtx, err := NewEnd2EndTestingContext(t)
+	require.NoError(t, err, "No error expected when creating testing context")
+
+	tCtx.CreateComputeTemplate(t)
+	t.Cleanup(func() {
+		tCtx.DeleteComputeTemplate(t)
+	})
+
+	tests := []End2EndTest{
+		{
+			Name:       "Create a valid cluster",
+			Endpoint:   "/apis/v1alpha2/namespaces/{namespace}/clusters",
+			HttpMethod: "POST",
+			InputJSON: `{
```

Comment:
Could it be strongly typed?

Reply:
Can you expand a bit on what do you mean by this? I was toying with the idea of having a golang http client with strongly typed methods. I am not sure if this what you are after.

Reply:
Yes, or possibly use the `github.com/ray-project/kuberay/proto/go_client` package at least to define the `clusterSpec` in a strongly-typed way, so it eases the development / maintenance, and have it serialised.

Reply:
Ok, this makes sense. I will go the route of adding a http client for ray_serve. It should make things a lot more cleaner.